Interaction with database

Danila-Peshev opened the pull request:
Added basic elements for working with the database, as well as endpoints and their processing.

Danila-Peshev committed:
commit:
Create base entities;
Create migrations.

Danila-Peshev committed:
Create based migrations

Danila-Peshev committed:
commit:
Add auth and user genre block

Danila-Peshev committed:
Add favoriteMovie block

Danila-Peshev committed:
Add watched movies block

Danila-Peshev committed:
Hidden some data

Danila-Peshev committed:
Formate code with Prettier

Isidora25 commented on src/auth/auth.controller.ts:
```diff
@@ -0,0 +1,13 @@
+export class AuthController {
+  constructor(private authService: AuthService) {}
+
+  @Post('/login')
```
это нам не понадобится, так как у нас graphql по тз, можно удалять

Isidora25 commented on src/auth/auth.service.ts:
```diff
@@ -0,0 +1,22 @@
+
+  private async validateUser(userDto: CreateUserDto) {
+    const user = await this.userService.findByEmail(userDto.email);
+    const passwordEquals = userDto.password === user?.password;
```
булевы переменные обычно начинаются со слова is

Isidora25 commented on src/auth/auth.service.ts:
```diff
@@ -0,0 +1,22 @@
+  private async validateUser(userDto: CreateUserDto) {
+    const user = await this.userService.findByEmail(userDto.email);
+    const passwordEquals = userDto.password === user?.password;
+    if (user && passwordEquals) {
+      return user;
+    }
+    throw new UnauthorizedException();
+  }
```
задача validateUser только проверить, что пароли совпадают 
пусть эксепшены выбрасывает метод логина, это его задача

Isidora25 commented on src/migrations/1724242104935-auto.ts:
название миграции - странное
через месяц по названию не вспомнишь, что тут происходит без детального изучения кода

Isidora25 commented on src/migrations/1724399693001-fill.ts:
то же самое замечание про имя

Isidora25 commented on src/typeorm.config.ts:
```diff
@@ -0,0 +1,25 @@
+config();
+
+const AppDataSource = new DataSource({
+  type: 'postgres',
+  host: process.env.DB_HOST,
+  port: Number(process.env.DB_PORT),
+  username: process.env.DB_USERNAME,
+  password: process.env.DB_PASSWORD,
+  database: process.env.DB_NAME,
+  entities: [join(__dirname, '**', '*.entity.{ts,js}')],
+  migrations: [join(__dirname, 'migrations', '*{.ts, .js}')],
```
а зачем тебе одно и то же в двух местах?

Danila-Peshev replied:
Один для миграций, один для взаимодействия в ходе работы
(Пытался вытащить connection данные в отдельный файл, но ts ругался когда я его использовал в TypeOrmModule)

Isidora25 commented on src/user-genre/userFavoriteGenre.controller.ts:
мы же вроде договаривались, что перенос сервисов будет в следующем пункте тз
плюс это rest, по тз нужен graphql

Isidora25 commented on src/user-genre/userFavoriteGenre.service.ts:
.

Isidora25 commented on src/user-movie/userFavoriteMovie.controller.ts:
.

Isidora25 commented on src/user-movie/userFavoriteMovie.entity.ts:
тут вроде и по контексту понятно, что favorite он чей-то, то есть юзера

Isidora25 commented on src/user-movie/userFavoriteMovie.service.ts:
.

Isidora25 commented on src/user-watched-movie/userWatchedMovie.controller.ts:
.

Isidora25 commented on src/user-watched-movie/userWatchedMovie.entity.ts:
```diff
@@ -0,0 +1,14 @@
+import { Entity, PrimaryGeneratedColumn, ManyToOne, Column } from 'typeorm';
+import { User } from '../user/user.entity';
+
+@Entity()
```
не нужно для этого заводить целую табличку, досточно булева флага в таблице movies

Isidora25 commented on src/user-watched-movie/userWatchedMovie.service.ts:
.

Danila-Peshev committed:
All interaction via requests has been removed

Danila-Peshev committed:
Commit:
I changed some paths for entities in the orm config, now migrations are successfully performed;
Now there is one essence of the movie with the isWatched flag.

Isidora25 commented on src/migrations/FillingInWithPrimaryData1724399693001.ts:
```diff
@@ -0,0 +1,39 @@
+            VALUES (1, 'ivan@mail.ru', 'ivan');
+        `);
+
+    await queryRunner.query(`
```
зачем тебе предзаполненные данные по фильмам?

Isidora25 commented on src/typeorm.config.ts:
```diff
@@ -0,0 +1,25 @@
+import { config } from 'dotenv';
+config();
+
+const AppDataSource = new DataSource({
```
конфиг для базы данных должен быть один 
представь, разработчику нужно внести изменение в этот кофиг и велика вероятность, что он не будет знать про второе место.
будет исправлять и ничего не поменяется

Danila-Peshev committed:
Commit:
Files were renamed in accordance with the norms;
A common configuration file has been created for the database and the data-source file;
The modules of films and genres have been renamed.

Danila-Peshev committed:
The filling of primary data in the tables of films and genres has been removed.

Isidora25 commented on src/data-source.ts:
```diff
@@ -0,0 +1,16 @@
+import { DataSource } from 'typeorm';
+import dataSourceOptions from './data-source-options.config';
+
+const AppDataSource = new DataSource({
```
оно тебе же вообще не нужно, у тебя есть typeOrm module, который отвечает за подключение к базе данных
